src/components/auth/recover/RecoveryKitButton.svelte

```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
+  import { goto } from '$app/navigation'
+
+  import { RECOVERY_STATES } from '$lib/auth/account'
+  import RightArrow from '$components/icons/RightArrow.svelte'
+  import Upload from '$components/icons/Upload.svelte'
+
+  export let handleFileInput: (files: FileList) => Promise<void>
+  export let state: RECOVERY_STATES
+
+  // Handle files uploaded directly through the file input
+  let files: FileList
+  $: if (files) {
+    handleFileInput(files)
+  }
+
+  $: buttonData = {
+    [RECOVERY_STATES.Processing]: {
+      text: 'Processing recovery kit...',
+      props: {
+        disabled: state === RECOVERY_STATES.Processing,
+        $$on_click: () => {}
```

Comment:
Ah interesting! What does the `$$` do?

Reply:
Nothing in particular 😅 i'm just using it here to denote that this is not a spreadable prop. I've seen it done in some other svelte projects 🤷🏼